src/datasources/asset-calibration/AssetCalibrationDataSource.ts

```diff
@@ -95,7 +105,14 @@ export class AssetCalibrationDataSource extends DataSourceBase<AssetCalibrationQ
   }
 
   createColumnNameFromDescriptor(field: FieldDTOWithDescriptor): string {
-    return field.columnDescriptors.map(descriptor => descriptor.value).join(' - ');
+    return field.columnDescriptors.map(descriptor => {
+      if (descriptor.type === ColumnDescriptorType.MinionId && this.state.systems) {
```

Comment:
Could you have a race condition in case the systems doesn't load fast enough?

Reply:
Yes it is. This is a issue I was thinking about.  I think this is an edge case.

The react whey is to use a hook that would re-render the component. But this class is not a react component.

The code I wrote will render the minion ID if the request takes to long.

We can also block the processing until the request finished. I dislike this option.

Reply:
Now it should be fix,